### ISSUE TITLE
zenodo and duecredit

### ISSUE THREAD

**Issue by dvolgyes** [fepegar/torchio#101]
Hi,

Scientific citation is always nice, but there is always a way to improve it. :)
First, you could deposit your code at zenodo.org.
It has a github integration, so all you need to do after connecting the services
is just creating a git release, and it will automatically deposited at Zenodo.
In this way people can refer to specific versions easily.
https://zenodo.org/

In general, Zenodo is very permissive and free of use (EU/CERN project).
You can deposit anything (source code, pdf, any files; by topic: finished papers, drafts, codes, data, etc.), and you can attach external DOIs, if your paper gets published later, you can use multiple versions, etc. 

Related: Duecredit is a lesser known but nice python library for embedding citation
information. The main idea that if you run a code with duecredit, it will collect
stat from all the imported projects, and emitt a list what you should cite.
So as a library author: you can annotate your main library with your citation information,
but you can also refer to implementations. E.g. if in function F use use a specific method,
you can annotate it. 

If your downstream users don't use duecredit, then  nothing happens, so it does not give
an extra dependency users. (Think about it like flake8 or coverage: if someone uses it, it gives information about the code, but it is not a depencency of the project.) 
https://github.com/duecredit/duecredit
But i think in CS we should lead in terms of automated credits, so i recommend duecredit.

**Comment by fepegar** [comment]
This repository has always been connected to Zenodo: https://zenodo.org/account/settings/github/repository/fepegar/torchio.

`duecredit` seems like an interesting project, but I'll close this issue for now.

**Comment by dvolgyes** [comment]
Zenodo: my bad, i missed it. 
But maybe you could put a link into the README.
(The above link is only visible to you, the record is: https://zenodo.org/record/3706851 )
Duecredit: fair enough, but it is nice, so i recommend to check it out later.

**Comment by fepegar** [comment]
Oh, sorry, thanks for correcting that link :)

I had it in the README until I published the paper, as that's my preferred way of acknowledging this library. Is that legit?

Are you part of the `duecredit` team?

**Comment by dvolgyes** [comment]
I am not part of duecredit, i just like their work.
Legit way: i think there is no standardized way.
At this moment, it is published, but not peer reviewed.
In a lot of sense, it counts similarly as oral communication, 
or a newspaper article.
If somebody uses the idea, should cite the paper.
If somebody uses the code, should cite the code too.

Usually, they are connected, but there are exceptions, like
scipy implements other people's algorithms,
so the citation would be "
The segmentation is performed with the Random Walker[1] algorithm
using SciPy[2] implementation".
[1] L. Grady: Random Walker
[2] Scipy...

So people often use one source. In case of arxiv, you can refer to the code
in arxiv paper (github link), and a citation would be enough like:
torchIO[arxiv id] v. 0.13.2. 
or could be 
torchIO[zenodo] v. 0.13.2
If you did not deposit the paper at zenodo, then probably it is easier to 
use the arxiv citation.

When you publish it at a peer review source, then in the peer reviewed paper
cite your current version, e.g. the source code (v.0.13.2) is accessible at [1].
[1] zenodo subversion (subDOI)
In this case you inform the reader which version is used in the paper,
and also zenodo displays other versions.
Science is all about informing the readers. So the reader than could
cite you like: 
-using original paper, original code, or 
- original paper, updated code, etc.
I don't like the unfortunately frequent practice when you need to cite a user's manual,
like for CUDA or Matlab. (E.g. see CUDA's citation where a user's manual is cited, but nobody actually provides a link to the manual. And i highly doubt that people still use CUDA 1.0, but everybody cites the 1.0.)

Long story short: at this moment, it doesn't really matter, if someone wants to cite the code, could even cite the git hash. But in final paper it is nice if you have your code deposited at a permanent storage and there is a link in the paper to the deposited version.
OK, i think i overexplained this. :)

**Comment by fepegar** [comment]
I celebrate your support to code acknowledgment!

I did not deposit the paper at Zenodo, because originally I wasn't even planning to write a paper. I see what you mean about idea vs code, but in this case, both are quite mixed. The paper mostly describes the library, it's not really some very novel method I came up with and for which I wrote and shared some code.

I like the idea of citing the current release in the paper. The plan is submitting to a peer-reviewed journal soon (I've been focusing on MICCAI in the past weeks), so I'll do that. Thanks for the advice.

I haven't fully understood all the ideas in your message, but thanks a lot for the explanations and for contributing. I'll take everything into consideration and I'll take a closer look at `duecredit` soon.

**Comment by dvolgyes** [comment]
Good luck with MICCAI!

**Comment by fepegar** [comment]
Thank you!

**Comment by fepegar** [comment]
Thanks again, @dvolgyes!
